tests/src/app/testqgsapplayoutvaliditychecks.cpp

```diff
@@ -222,13 +222,12 @@ void TestQgsLayoutValidityChecks::testPictureValidity()
   QgsLayoutValidityCheckContext context( &l );
   QgsFeedback f;
 
-  // invalid picture source
+  // invalid picture source, will load ":/images/composer/missing_image.svg"
   picture->setPicturePath( QStringLiteral( "blaaaaaaaaaaaaaaaaah" ) );
   QgsLayoutPictureSourceValidityCheck check;
   QVERIFY( check.prepareCheck( &context, &f ) );
   QList< QgsValidityCheckResult > res = check.runCheck( &context, &f );
-  QCOMPARE( res.size(), 1 );
-  QCOMPARE( res.at( 0 ).type, QgsValidityCheckResult::Warning );
+  QCOMPARE( res.size(), 0 );
```

Comment:
@nyalldawson how does that look to you?

Reply:
No, the original result was correct

Reply:
When we move IMAGE_RCCS to qgis_core, QString badFile (QStringLiteral (": /images/composer/missing_image.svg")) in QgsLayoutItemPicture :: loadPicture will return a valid value. This is the reason for the different results.
```
void QgsLayoutItemPicture::loadPicture( const QVariant &data )
{
  ...
  if ( mMode != FormatUnknown ) //make sure we start with a new QImage
  {
    recalculateSize();
  }
  else if ( mHasExpressionError || !mEvaluatedPath.isEmpty() )
  {
    //trying to load an invalid file or bad expression, show cross picture
    mMode = FormatSVG;
    mIsMissingImage = true;
    QString badFile( QStringLiteral( ":/images/composer/missing_image.svg" ) );
    mSVG.load( badFile );
    if ( mSVG.isValid() )
    {
      mMode = FormatSVG;
      QRect viewBox = mSVG.viewBox(); //take width/height ratio from view box instead of default size
      mDefaultSvgSize.setWidth( viewBox.width() );
      mDefaultSvgSize.setHeight( viewBox.height() );
      recalculateSize();
    }
  }
}
```

Reply:
Thanks -- you'll need https://github.com/qgis/QGIS/pull/35864 to fix this